bootable-jar/WFLY-13305_Bootable_Jar.adoc

```diff
@@ -0,0 +1,450 @@
+* User make changes
+* mvn package ⇒ full repackaging
+* ...
+
+Dev mode, server built/started once, app automatically re-deployed.
+
+* mvn wildfly-jar:dev ⇒ hollow server built and started
+* mvn package -Ddev ⇒ Fast, no packaging, app copied to deployments dir. Application automatically deployed.
+* User make changes
+* mvn package -Ddev ⇒ Fast, no packaging, app copied to deployments dir. Application automatically re-deployed.
+* ...
+* mvn wildfly-jar:shutdown
+* User is fine with his changes, he can do the server+app packaging: mvn package
+
+NB: This workflow doesn’t require support in IDE, it is 100% Maven. It could be optimized with IDE plugin (e.g.: netbeans plugin to track static 
+files and avoid to re-package in this case).
+
+
+==== Cloud context
+
+The plugin allows to generate a bootable JAR usable in various Cloud execution contexts: custom container, Google JIB, Fabric8, Java s2i build.
+
+As an example, s2i binary build of the https://github.com/wildfly-extras/wildfly-jar-maven-plugin/blob/master/examples/mp-config/README.md[microprofile-config example] is provided. 
+
+
+===== Configuring the Maven plugin for Cloud environment
+
+The Maven plugin configuration item ```<cloud></cloud>``` allows to build a bootable JAR for cloud environment. By default the server is configured to run inside an OpenShift context.
+Set the cloud child element ```<type>openshift|kubernetes</type>``` to select the targeted cloud platform.
+
+The sever configuration is updated in order to properly operate in a cloud environment:
+
+* The ```microprofile-health``` and ```core-tools``` (that contains WildFly CLI) galleon layers are provisioned. They are required for the  OpenShift probes and WildFly OpenShift operator to properly operate.
+* The public and private inet addresses are bound to the value of the ```HOSTNAME``` environment variable if defined (defined in OpenShift PODS).
+* The management inet address is bound to the 0.0.0.0 inet address allowing for local (required by WildFly CLI) and remote access (required by OpenShift readiness and liveness probes).
+* The transaction subsystem id is set to the value of ```jboss.node.name```.
+* The ```jboss.node.name``` system propery, if not set, is set to the value of ```HOSTNAME``` environment variable if defined (defined in OpenShift PODS). The node name value
+is truncated to a max of 23 characters in order for the transaction subsystem to properly operate.
+* The server logs are printed in the console.
+* jgroups subsystem is configured to use kubernetes.KUBE_PING jgroups protocol for both tcp (default stack) and udp. PING and MPING protocols are removed.
+* It is possible to configure jgroups to use un-encrypted password authentication. Set the ```<cloud>``` child element ```<enable-jgroups-password>true|false</enable-jgroups-password>``` to enable authentication. 
+NB: When authentication is enabled, the environment variable ```JGROUPS_CLUSTER_PASSWORD``` must be set.
```

Comment:
What happens if is not set? Is there any logged message or event?

Reply:
The server will fail to start with traces saysing that the expression can't be resolved.

Reply:
Ok, great, can this behavior be mentioned explicitly?